File: ems-cloud-mgmt-sdk/clojure/src/ems_rest_api/specs/balance_data_inner.clj
(ns ems-rest-api.specs.balance-data-inner
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def balance-data-inner-data
  {
   (ds/opt :asset_id_exchange) string?
   (ds/opt :asset_id_coinapi) string?
   (ds/opt :balance) float?
   (ds/opt :available) float?
   (ds/opt :locked) float?
   (ds/opt :last_updated_by) string?
   (ds/opt :rate_usd) float?
   (ds/opt :traded) float?
   })

(def balance-data-inner-spec
  (ds/spec
    {:name ::balance-data-inner
     :spec balance-data-inner-data}))

File: ems-gateway-rest-sdk/javascript-closure-angular/API/Client/BalanceDataInner.js
goog.provide('API.Client.Balance_data_inner');

/**
 * @record
 */
API.Client.BalanceDataInner = function() {}

/**
 * Exchange currency code.
 * @type {!string}
 * @export
 */
API.Client.BalanceDataInner.prototype.assetIdExchange;

/**
 * CoinAPI currency code.
 * @type {!string}
 * @export
 */
API.Client.BalanceDataInner.prototype.assetIdCoinapi;

/**
 * Value of the current total currency balance on the exchange.
 * @type {!number}
 * @export
 */
API.Client.BalanceDataInner.prototype.balance;

/**
 * Value of the current available currency balance on the exchange that can be used as collateral.
 * @type {!number}
 * @export
 */
API.Client.BalanceDataInner.prototype.available;

/**
 * Value of the current locked currency balance by the exchange.
 * @type {!number}
 * @export
 */
API.Client.BalanceDataInner.prototype.locked;

/**
 * Source of the last modification. 
 * @type {!string}
 * @export
 */
API.Client.BalanceDataInner.prototype.lastUpdatedBy;

/**
 * Current exchange rate to the USD for the single unit of the currency. 
 * @type {!number}
 * @export
 */
API.Client.BalanceDataInner.prototype.rateUsd;

/**
 * Value of the current total traded.
 * @type {!number}
 * @export
 */
API.Client.BalanceDataInner.prototype.traded;

/** @enum {string} */
API.Client.BalanceDataInner.LastUpdatedByEnum = { 
  INITIALIZATION: 'INITIALIZATION',
  BALANCE_MANAGER: 'BALANCE_MANAGER',
  EXCHANGE: 'EXCHANGE',
}
